gnomad/utils/filtering.py

```diff
@@ -533,40 +533,46 @@ def split_vds_by_strata(
     }
 
 
-def filter_freq_by_meta(
-    freq_expr: hl.expr.ArrayExpression,
-    freq_meta_expr: hl.expr.ArrayExpression,
+def filter_arrays_by_meta(
+    meta_expr: hl.expr.ArrayExpression,
+    meta_indexed_exprs: Dict[str, hl.expr.ArrayExpression],
     items_to_filter: Union[Dict[str, List[str]], List[str]],
     keep: bool = True,
     combine_operator: str = "and",
-) -> Tuple[hl.expr.ArrayExpression, hl.expr.ArrayExpression]:
+) -> Tuple[hl.expr.ArrayExpression, Dict[str, hl.expr.ArrayExpression]]:
     """
-    Filter frequency and frequency meta expressions specified by `items_to_filter`.
+    Filter both metadata array expression and meta data indexed expression by `items_to_filter`.
 
     The `items_to_filter` can be used to filter in the following ways based on
-    `freq_meta_expr` items:
+    `meta_expr` items:
     - By a list of keys, e.g. ["sex", "downsampling"].
     - By specific key: value pairs, e.g. to filter where 'pop' is 'han' or 'papuan'
     {"pop": ["han", "papuan"]}, or where 'pop' is 'afr' and/or 'sex' is 'XX'
     {"pop": ["afr"], "sex": ["XX"]}.
 
-    The items can be kept or removed from `freq_expr` and `freq_meta_expr` based on the
-    value of `keep`.
+    The items can be kept or removed from `meta_indexed_expr` and `meta_expr` based on
+    the value of `keep`. For example if `meta_indexed_exprs` is {'freq': ht.freq,
+'freq_meta_sample_count': ht.index_globals().freq_meta_sample_count} and `meta_expr`
```

Comment:
sorry, you have to add the indentation. I forgot to copy that, like the , from last time.